subsys/bluetooth/mesh/Kconfig

```diff
@@ -128,6 +128,17 @@ choice TFM_PROFILE_TYPE
 	default TFM_PROFILE_TYPE_NOT_SET
 endchoice
 
+config BT_MESH_DFU_METADATA_ON_BUILD
+	bool "Generate DFU metadata on build"
+	default y if BT_MESH_DFU_SRV || BT_MESH_DFU_CLI
+	depends on SIGN_IMAGES
+	depends on BT_MESH_DFU_METADATA
```

Comment:
Minor: Then BT_MESH_DFU_METADATA should also be "default y if (BT_MESH_DFU_SRV || BT_MESH_DFU_CLI)". Why should BT_MESH_DFU_METADATA explicit be set to y if this is enabled by default and we are dependent on it.